builder/dockerfile/internals.go

```diff
@@ -248,7 +248,6 @@ func (b *Builder) download(srcURL string) (fi builder.FileInfo, err error) {
 	progressReader := progress.NewProgressReader(resp.Body, progressOutput, resp.ContentLength, "", "Downloading")
 	// Download and dump result to tmp file
 	if _, err = io.Copy(tmpFile, progressReader); err != nil {
-		tmpFile.Close()
```

Comment:
Thanks @cyphar 
I think using a defer is a good habit in golang. And this case is somewhat particular. In my opinion, I am not on the side of close a fd twice, although it will not report error. As a result, I prefer to choose to add more Close() which seems some redundant, pity. While it will still have no advantages of defer.